src/book/models/InvestmentAccount.ts

```diff
@@ -280,18 +280,30 @@ export default class InvestmentAccount {
    */
   _dividend(split: Split): void {
     split.transaction.splits.forEach(otherSplit => {
-      if (split.guid !== otherSplit.guid && otherSplit.quantity > 0) {
+      if (
+        split.guid !== otherSplit.guid
+        && this.currency === otherSplit.account.commodity.mnemonic
+        && otherSplit.quantity > 0
```

Comment:
instead of selecting a random account with different guid, we pick the account that matches the currency of this account